networks/p2p/server.go

```diff
@@ -787,6 +787,21 @@ running:
 	}
 }
 
+// Stop terminates the server and all active peer connections.
+// It blocks until all active connections are closed.
+func (srv *MultiChannelServer) Stop() {
+	for _, listener := range srv.listeners {
```

Comment:
How about below code instead of call baseServer.Stop() for lock/unlock and close quit/wait?
```suggestion
	srv.lock.Lock()
	defer srv.lock.Unlock()
	if !srv.running {
		return
	}
	srv.running = false
	for _, listener := range srv.listeners {
			listener.Close()
	}
	close(srv.quit)
	srv.loopWG.Wait()
```

Reply:
Already reflected.